src/main/telemetry/ibus.c

```diff
@@ -0,0 +1,430 @@
+ * voltage leaving 0x1 to 0xF remaining.
+ *
+ * Having learned it's address, the sensor simply echos the message back:
+ *
+ *
+ *                                                      /--------\
+ *                              Yes, i'm here, hello! < | Sensor |
+ *                       [ 0x04, 0x81, 0x7A, 0xFF ]     \--------/
+ *
+ *     0x04, 0x81, 0x7A, 0xFF - Echo back received packet
+ *
+ *
+ * On receipt of a response, the iBus rx next requests the sensor's type:
+ *
+ *
+ *  /---------\
+ *  | IBUS RX | > Sensor at address 1, what type are you?
+ *  \---------/     [ 0x04, 0x91, 0x6A, 0xFF ]
+ *
+ *     0x04       - Packet Length
+ *     0x91       - bits 7-4 Command (1001 = request sensor type)
+ *                  bits 3-0 Address (0001 = address 1)
+ *     0x6A, 0xFF - Checksum, 0xFFFF - (0x04 + 0x91)
+ *
+ *
+ * To which the sensor responds with its details:
+ *
+ *
+ *                                                      /--------\
+ *                              Yes, i'm here, hello! < | Sensor |
+ *                [ 0x06 0x91 0x00 0x02 0x66 0xFF ]     \--------/
+ *
+ *     0x06       - Packet Length
+ *     0x91       - bits 7-4 Command (1001 = request sensor type)
+ *                  bits 3-0 Address (0001 = address 1)
+ *     0x00       - Measurement type (0 = internal voltage)
+ *     0x02       - Unknown, always 0x02
+ *     0x66, 0xFF - Checksum, 0xFFFF - (0x06 + 0x91 + 0x00 + 0x02)
+ *
+ *
+ * The iBus rx continues the discovery process by querying the next
+ * address. Discovery stops at the first address which does not respond.
+ *
+ * The iBus rx then begins a continual loop, requesting measurements from
+ * each sensor discovered:
+ *
+ *
+ *  /---------\
+ *  | IBUS RX | > Sensor at address 1, please send your measurement
+ *  \---------/     [ 0x04, 0xA1, 0x5A, 0xFF ]
+ *
+ *     0x04       - Packet Length
+ *     0xA1       - bits 7-4 Command (1010 = request measurement)
+ *                  bits 3-0 Address (0001 = address 1)
+ *     0x5A, 0xFF - Checksum, 0xFFFF - (0x04 + 0xA1)
+ *
+ *
+ *                                                      /--------\
+ *                                I'm reading 0 volts < | Sensor |
+ *                [ 0x06 0xA1 0x00 0x00 0x5E 0xFF ]     \--------/
+ *
+ *     0x06       - Packet Length
+ *     0xA1       - bits 7-4 Command (1010 = request measurement)
+ *                  bits 3-0 Address (0001 = address 1)
+ *     0x00, 0x00 - The measurement
+ *     0x58, 0xFF - Checksum, 0xFFFF - (0x06 + 0xA1 + 0x00 + 0x00)
+ *
+ *
+ * Due to the limited telemetry data types possible with ibus, we
+ * simply send everything which can be represented. Currently this
+ * is voltage and temperature.
+ *
+ */
+
+
+PG_REGISTER_WITH_RESET_TEMPLATE(ibusTelemetryConfig_t, ibusTelemetryConfig, PG_IBUS_TELEMETRY_CONFIG, 0);
+
+PG_RESET_TEMPLATE(ibusTelemetryConfig_t, ibusTelemetryConfig,
+                  .report_cell_voltage = false,
+                 );
+
+#define IBUS_TASK_PERIOD_US (500)
```

Comment:
@inode 
Regarding the cpu load, you can try to increase this value in steps until you see the telemetry stop working and then back off a bit until it is working again. Then see if you can get GPS online again.